interfaces/builtin/mount_control.go

```diff
@@ -345,15 +359,26 @@ func validateMountTypes(types []string) error {
 	return nil
 }
 
-func validateMountOptions(options []string) error {
-	if len(options) == 0 {
+func validateMountOptions(mountInfo *MountInfo) error {
+	if len(mountInfo.options) == 0 {
 		return errors.New(`mount-control "options" cannot be empty`)
 	}
-	for _, o := range options {
-		if !strutil.ListContains(allowedMountOptions, o) {
-			return fmt.Errorf(`mount-control option unrecognized or forbidden: %q`, o)
+	// With "functionfs" none of the valid "options" can be harmful so no need to check against allowedMountOptions
+	if mountInfo.isType("functionfs") {
+		if err := validateNoAppArmorRegexpWithError(`cannot use mount-control "option" attribute`, mountInfo.options...); err != nil {
+			return err
+		}
+	} else {
+		for _, o := range mountInfo.options {
+			if !strutil.ListContains(allowedMountOptions, o) {
+				return fmt.Errorf(`mount-control option unrecognized or forbidden: %q`, o)
+			}
 		}
 	}
+	fsExclusiveOption := optionIncompatibleWithFsType(mountInfo.options)
```

Comment:
has we don't use allowedMountOptions for functionfs the comment at https://github.com/snapcore/snapd/blob/ba042ca8bbfb695db2f72a4668c3ee5e3b436377/interfaces/builtin/mount_control.go#L100 does not hold, I'm not sure what are the consequences but it seems like we do need to expand the list there to cover things that would otherwise be impossible?

Reply:
@pedronis @mvo5 

Expanded `optionsWithoutFsType`.
Added more options to make the list complete.